spec/src/main/asciidoc/overview.asciidoc

```diff
@@ -0,0 +1,90 @@
+
+Goals
+
+    - Proper management of CompletableFuture threads by the container.
+
+    - Mechanism for thread context propagation to CompletableFuture actions that reduces the need for boilerplate code.
+
+    - Full compatibility with EE Concurrency spec, such that proposed interfaces can eventually be seamlessly merged into EE Concurrency.
+
+=== Solution
+
+This specification introduces two interfaces that contain methods that we hope will eventually be added to Jakarta EE Concurrency.
+
+The interface, org.eclipse.microprofile.concurrent.ManagedExecutor, provides methods for obtaining managed instances of CompletableFuture which are backed by the managed executor as the default asynchronous execution facility and the default mechanism of defining thread context propagation. Similar to EE Concurrency’s ManagedExecutorService, the MicroProfile ManagedExecutor also implements the Java SE java.util.concurrent.ExecutorService interface, using managed threads when asynchronous invocation is required and disallowing the same life cycle methods as ManagedExecutorService. It is intended that ManagedExecutor methods will one day be added to ManagedExecutorService, and for a single implementation to be capable of simultaneously implementing both interfaces, both currently as well as after adoption into Jakarta EE.
+
+A second interface, org.eclipse.microprofile.concurrent.ThreadContext, provides methods for individually contextualizing dependent stage actions. This gives the user finer-grained control over the capture and propagation of thread context. It is intended that ThreadContext methods will one day be added to EE Concurrency’s ContextService and for a single implementation to be capable of simultaneously implementing both interfaces, both currently as well as after adoption into Jakarta EE.
+
+=== Builders
+
+Instances of ManagedExecutor and ThreadContext can be constructed via builders with fluent API, for example,
+
+[source, java]
+----
+    ManagedExecutor executor = ManagedExecutor.builder()
+        .propagated(ThreadContext.APPLICATION)
+        .maxAsync(5)
+        .build();
+
+    ThreadContext threadContext = ThreadContext.builder()
+        .propagated(ThreadContext.APPLICATION, ThreadContext.CDI)
+        .cleared(ThreadContext.ALL_REMAINING)
+        .build();
+----
+
+Applications should shut down ManagedExecutors that they build after they are no longer needed. The shutdown request serves as a signal notifying the container that resources can be safely cleaned up.
+
+=== Injection
+
+Instances of ManagedExecutor and ThreadContext can be injected into CDI beans via the `@Inject` annotation, for example:
+
+[source, java]
+----
+    @Inject ManagedExecutor executor;
```

Comment:
Have we actually specified how this "default" instance looks like?
E.g. when no config is present whatsoever.

Reply:
Good point, we definitely need to do that (and the same for the default instance of ThreadContext).  For now, I'll document it as being equivalent to using the builder with all of the defaults, ManagedExecutor.builder().build()